README.md

```diff
@@ -124,6 +148,14 @@ Because of their generality, vouchers can be used in a wide range of application
 | ERC-1155 | Token contract | `safeTransferFrom(address,address,uint256,uint256,data)` [:page_facing_up:](https://eips.ethereum.org/EIPS/eip-1155#specification) |
 | ERC-1155 | Token contract | `safeBatchTransferFrom(address,address,uint256[],uint256[],data)` [:page_facing_up:](https://eips.ethereum.org/EIPS/eip-1155#specification) [^3] |
 
+Please note that the voucher payload should be encoded according to the [Ethereum ABI specification for calling contract functions](https://docs.soliditylang.org/en/v0.8.19/abi-spec.html). As such, it should start with the first four bytes of the Keccak-256 hash of the function selector string (as given in the table above), followed by the ABI-encoded parameter values.
+
+As an example, the voucher for a simple ERC-20 transfer (2nd line in the table above) to address `0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266` with amount 100 should specify the following payload:
```

Comment:
We could rephrase this to "a simple ERC-20 transfer (...) of 100 tokens to address `...` should [...]"

Reply:
I see your point now. Thanks for the clarification! :-)

Reply:
Oh I was hoping you would have a great suggestion as to how to express this better!

Reply:
Well, clearly the phrase "Bob has $x$ tokens" is ambiguous, as our interpretations have diverged. Does it mean $x$ or $x / 10^D$ [^1]?

**In my interpretation**, if you say Bob has 42 tokens, then I will think that if you call `balanceOf(bob)`, then you'll get the `uint256` value of 42 exactly. Now, for the user, this number may be displayed differently depending on the number of decimal places. So, if token FOO uses, let's say, 2 decimal places, then it would appear to me as 0.42 FOO, but in reality I have 42 tokens.

**Another interpretation** (which is probably that of most users) is that `balanceOf(bob)` would return $42*10^D$. In this case, are users even aware they are actually dealing with discreet tokens, and that EIP-20 doesn't deal with floating point numbers? I don't know if they have a name for the discreet tokens. If so, we should use this name.

[^1]: where D is the number of decimal places

Reply:
I think the correct way of saying is 42 tokens or 42e18 amount. Just like if you say you are transfering 42 ethers you .

So if bob has 42 tokens, balanceOf should return 42e18.

I would write something "a voucher for 100 tokens would have as parameters (...) 100e18" with a footnote explaining that the parameter is in the undivisible unit.

Reply:
Thanks for the inputs guys. And all because I want a small number (0x64) to appear in the payload! Since this PR was approved I am closing it as is, we can improve later on.